supervisor/backups/backup.py

```diff
@@ -502,28 +502,32 @@ async def open(self, location: str | None | type[DEFAULT]) -> AsyncGenerator[Non
             if location == DEFAULT
             else self.all_locations[location][ATTR_PATH]
         )
-        if not backup_tarfile.is_file():
-            self.sys_create_task(self.sys_backups.reload(location))
-            raise BackupFileNotFoundError(
-                f"Cannot open backup at {backup_tarfile.as_posix()}, file does not exist!",
-                _LOGGER.error,
-            )
 
         # extract an existing backup
-        self._tmp = TemporaryDirectory(dir=str(backup_tarfile.parent))
-
         def _extract_backup():
-            """Extract a backup."""
+            if not backup_tarfile.is_file():
+                raise BackupFileNotFoundError(
+                    f"Cannot open backup at {backup_tarfile.as_posix()}, file does not exist!",
+                    _LOGGER.error,
+                )
+            self._tmp = TemporaryDirectory(dir=str(backup_tarfile.parent))
```

Comment:
```suggestion
            return TemporaryDirectory(dir=str(backup_tarfile.parent))
```
Probably overcautious but I've just been bitten by this a lot with hard to debug race issues. As long as its an easy change I always prefer to simply return from the executor and do actual modification of instance data in the asyncio loop. We can then set `self._tmp` to the value of this return outside.